atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DdlConfig.java

```diff
@@ -15,17 +15,23 @@
  */
 package com.palantir.atlasdb.keyvalue.dbkvs;
 
+import static com.fasterxml.jackson.annotation.JsonTypeInfo.As;
+import static com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+
 import org.immutables.value.Value;
 
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.base.Supplier;
 import com.palantir.atlasdb.AtlasDbConstants;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.DbTableFactory;
+
+@JsonTypeInfo(use = Id.NAME, include = As.PROPERTY, property = "type", visible = false)
+public abstract class DdlConfig {
```

Comment:
What value do each of the instantiation options provide?  Seems like unless we are going to override them with defaults per kvs type they aren't necessary.